app/src/main/java/org/openedx/app/MainFragment.kt

```diff
@@ -81,11 +83,20 @@ class MainFragment : Fragment(R.layout.fragment_main) {
             }
         }
 
-        requireArguments().apply {
-            this.getString(ARG_COURSE_ID, null)?.apply {
-                router.navigateToCourseDetail(parentFragmentManager, this)
+        requireArguments().run {
```

Comment:
Apologies for the oversight. Since we are not returning any value, we can proceed with using `apply`, as we are updating the bundle properties after processing them.

@k1rill thorughs?